test/dist_autograd_test.py

```diff
@@ -705,6 +705,24 @@ def test_context_cleanup_many_workers(self):
         success = _all_contexts_cleaned_up()
         self.assertTrue(success)
 
+    @dist_init
+    def test_context_cleanup_tensor_no_grad(self):
```

Comment:
nit: would be nice to add some docs explaining what issue the unit test is catching